homeassistant/components/onkyo/receiver.py

```diff
@@ -0,0 +1,33 @@
+"""Onkyo receiver."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+import pyeiscp
+
+if TYPE_CHECKING:
+    from .media_player import OnkyoMediaPlayer
+
+
+@dataclass
+class Receiver:
+    """Onkyo receiver."""
+
+    conn: pyeiscp.Connection
+    model_name: str
+    identifier: str
+    name: str
+    discovered: bool
+    entities: dict[str, OnkyoMediaPlayer] = field(default_factory=dict)
```

Comment:
We should not have the entities outside of their module. This is a flaw that the current media player has, and I'd rather see that refactored so the objects aren't touched anymore after creating than typing it

Reply:
I agree, I removed it.

There is one problem though. The service `onkyo_select_hdmi_output` is not a proper entity service. It is registered directly under the media_player key. But it needs access to the entities, so for now I'm going to put them on a separate key in `hass.data`. Once we deprecate and replace the service, we can remove that workaround.

Reply:
But now we're still refactoring the way we store them, and that's what I want to avoid

Reply:
It has to happen.

Now we access them directly from closure in `async_setup_platform`. But that is going away with the config flow. We will have only `async_setup_entry` and no way to do the same.

We have to put the entities somewhere, so that the old service can access them.